source/task_scheduler.cpp

```diff
@@ -332,12 +332,21 @@ void TaskScheduler::Run(uint fiberPoolSize, TaskFunction mainTask, void *mainTas
 }
 
 void TaskScheduler::AddTask(Task task, AtomicCounter *counter) {
+	static thread_local std::size_t externalIndex = 0;
+
 	if (counter != nullptr) {
 		counter->Store(1);
 	}
 
 	const TaskBundle bundle = {task, counter};
-	m_tls[GetCurrentThreadIndex()].TaskQueue.Push(bundle);
+	std::size_t const threadIdx = GetCurrentThreadIndex();
+	if (threadIdx != FTL_INVALID_INDEX) {
+		m_tls[GetCurrentThreadIndex()].TaskQueue.Push(bundle);
+	}
+	else {
+		m_tls[externalIndex].TaskQueue.Push(bundle);
```

Comment:
TaskQueue.Push isn't thread safe. The internal algorithm has strong assumptions that there is a single pusher, and multiple stealers.